File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        int n=strs.size();
        string ans="";
        string a=strs[0];
        string z=strs[n-1];
        for(int i=0;i<min(a.size(),z.size());i++){
            if(a[i]!=z[i]){
                return ans;
            }
        ans.push_back(a[i]);
        }
     return ans;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        ans=""
        if not strs:
            return ""
        first,last=strs[0],strs[-1]
        n=min(len(first),len(last))
        for i in range(n):
            if (first[i]==last[i]):
                ans+=first[i]
            else:
                break
        return ans



        